core/src/saros/negotiation/additional_resource_data/AbstractPossibleRepresentationProvider.java

```diff
@@ -0,0 +1,162 @@
+
+    String flattenedPossibleRepresentations =
+        possibleRepresentations
+            .stream()
+            .map(
+                pair ->
+                    ESCAPE_INNER.escape(pair.getLeft())
+                        + DELIMITER_INNER
+                        + ESCAPE_INNER.escape(pair.getRight()))
+            .map(ESCAPE_OUTER::escape)
+            .collect(Collectors.joining(DELIMITER_OUTER));
+
+    return Collections.singletonMap(POSSIBLE_REPRESENTATIONS_KEY, flattenedPossibleRepresentations);
+  }
+
+  /**
+   * Returns the possible representations contained in the given resource negotiation data.
+   *
+   * @param resourceNegotiationData the resource negotiation data
+   * @return the possible representations contained in the given resource negotiation data or an
+   *     empty list of no such representations are specified by the negotiation data
+   */
+  public static List<Pair<String, String>> getPossibleRepresentations(
+      ResourceNegotiationData resourceNegotiationData) {
+    String flattenedPossibleRepresentation =
+        resourceNegotiationData.getAdditionalResourceData().get(POSSIBLE_REPRESENTATIONS_KEY);
+
+    if (flattenedPossibleRepresentation == null) {
+      return Collections.emptyList();
+    }
+
+    return split(flattenedPossibleRepresentation);
+  }
+
+  /**
+   * Splits and unescapes given flattened possible representation string. Returns the possible
+   * representations contained in the given string.
+   *
+   * @param flattenedPossibleRepresentations the flattened possible representation string contained
+   *     in the resource negotiation data
+   * @return the possible representations contained in the given string
+   */
+  private static List<Pair<String, String>> split(String flattenedPossibleRepresentations) {
+    return ESCAPE_OUTER
+        .getSplitPattern()
+        .splitAsStream(flattenedPossibleRepresentations)
+        .map(ESCAPE_OUTER::unescape)
+        .map(
+            joinedPossibleRepresentation -> {
+              String[] values = ESCAPE_INNER.getSplitPattern().split(joinedPossibleRepresentation);
+
+              if (values.length != 1 && values.length != 2) {
+                throw new IllegalStateException(
+                    "Wrong number of split results - joined string: "
+                        + joinedPossibleRepresentation
+                        + ", split result: "
+                        + Arrays.toString(values));
+              }
+
+              String name = values[0];
+
+              String path;
+              if (values.length == 2) {
+                path = values[1];
+              } else {
+                path = "";
+              }
+
+              return new ImmutablePair<>(ESCAPE_INNER.unescape(name), ESCAPE_INNER.unescape(path));
+            })
+        .collect(Collectors.toList());
+  }
+
+  /** Utility class to escape, unescape, and split strings. */
+  private static class EscapeUtils {
```

Comment:
This is somewhat generic. I already used this when remembering project selections. Can you please move it to the util package ?

Reply:
Done. Moved to class EscapeUtil.